docs/strategies/circuit-breaker.md

```diff
@@ -91,23 +94,62 @@ new ResiliencePipelineBuilder<HttpResponseMessage>().AddCircuitBreaker(optionsSt
+| `StateProvider`          | `null`                                                  | If provided then the circuit's current state can be retrieved via a `CircuitBreakerStateProvider` object.                                                 |
+| `OnClosed`               | `null`                                                  | If provided then it will be invoked after the circuit transitions to either the `Closed` or `Isolated` states.                                            |
+| `OnOpened`               | `null`                                                  | If provided then it will be invoked after the circuit transitions to the `Opened` state.                                                                  |
+| `OnHalfOpened`           | `null`                                                  | If provided then it will be invoked after the circuit transitions to the `HalfOpened` state.                                                              |
 
 > [!NOTE]
 > If both `BreakDuration` and `BreakDurationGenerator` are specified then `BreakDuration` will be ignored.
 
+---
+
+> [!IMPORTANT]
+> If the `MinimumThroughput` is not reached during the `SamplingDuration` then the `FailureRatio` is ignored.
+> In other words, the circuit will not break even if all of the executions failed when their quantity is below the minimum throughput.
+
+## Telemetry
+
+The circuit breaker strategy reports the following telemetry events:
+
+| Event Name            | Event Severity | When?                                                      |
+|-----------------------|----------------|------------------------------------------------------------|
+| `OnCircuitClosed`     | `Information`  | Just before the strategy calls the `OnClosed` delegate     |
+| `OnCircuitOpened`     | `Error`        | Just before the strategy calls the `OnOpened` delegate     |
+| `OnCircuitHalfOpened` | `Warning`      | Just before the strategy calls the `OnHalfOpened` delegate |
+
+Here are some sample events:
+
+```none
+Resilience event occurred. EventName: 'OnCircuitOpened', Source: 'MyApplication/MyTestPipeline/MyCircuitBreakerStrategy', Operation Key: 'MyCircuitedOperation', Result: 'Exception of type 'CustomException' was thrown.'
+    CustomException: Exception of type 'CustomException' was thrown.
+        at Program.<>c.<<Main>b__0_1>d.MoveNext()
+        ...
+        at Polly.ResiliencePipeline.<>c__8`1.<<ExecuteAsync>b__8_0>d.MoveNext() in /_/src/Polly.Core/ResiliencePipeline.AsyncT.cs:line 95
+
+Resilience event occurred. EventName: 'OnCircuitHalfOpened', Source: 'MyApplication/MyTestPipeline/MyCircuitBreakerStrategy', Operation Key: 'MyCircuitedOperation', Result: ''
+
+Resilience event occurred. EventName: 'OnCircuitClosed', Source: 'MyApplication/MyTestPipeline/MyCircuitBreakerStrategy', Operation Key: 'MyCircuitedOperation', Result: '42'
```

Comment:
```suggestion
Resilience event occurred. EventName: 'OnCircuitClosed', Source: 'MyPipeline/MyPipelineInstance/MyCircuitBreakerStrategy', Operation Key: 'MyCircuitedOperation', Result: '42'
```

Would better represent what those values actually are. Applies to other strategies as well.

Reply:
Done